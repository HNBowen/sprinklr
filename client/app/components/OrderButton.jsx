import React from 'react'

const OrderButton = (props) => {
  return (
      <button onClick={props.handleClick}>Order Plants</button>
    )
}

export default OrderButton