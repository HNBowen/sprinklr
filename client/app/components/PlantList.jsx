import React from 'react'

import PlantTile from './PlantTile.jsx'

const PlantList = (props) => {
  return (
      <div>
        {props.plants.map(function(plant) {
          return <PlantTile plant={plant} key={plant.id} handleClick={props.handlePlantTileClick}/>
        })}
      </div>
    )
}

export default PlantList