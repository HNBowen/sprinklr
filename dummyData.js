export const existingPlants = [
  {
    name: "Peace Lily",
    img: "https://cdn.nurserylive.com/images/stories/virtuemart/product/resized/nurserylive-Peace-Lily,-Spathiphyllum-Plant_128x128.jpg",
    lastWatered: new Date()
  },
  {
    name: "Monstera",
    img: "https://static.wixstatic.com/media/e7be30_b5577954ef5347409f7c4ebaeadce8ce~mv2_d_4288_2848_s_4_2.jpg/v1/fill/w_256,h_256,q_85,usm_0.66_1.00_0.01/e7be30_b5577954ef5347409f7c4ebaeadce8ce~mv2_d_4288_2848_s_4_2.jpg",
    lastWatered: new Date()
  }
];

export const plantsToAdd = [
  {
    name: "Pothos",
    img: "https://images-na.ssl-images-amazon.com/images/I/71C+qx6FiUL._CR412,0,1060,1060_UX128.jpg",
    lastWatered: new Date()
  },
  {
    name: "Fiddle Leaf",
    img: "https://i.pinimg.com/236x/6f/d3/2e/6fd32e64735e460852ec3c507df10354.jpg",
    lastWatered: new Date()
  }
]

