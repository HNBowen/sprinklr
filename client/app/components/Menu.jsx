import React from 'react'

import OrderButton from './OrderButton.jsx'
import AddPlantButton from './AddPlantButton.jsx'

const Menu = (props) => {
  return (
      <div>
        <OrderButton handleClick={props.handleOrderButtonClick}/>
        <AddPlantButton handleClick={props.handleAddPlantButtonClick}/>
      </div>
    )
}

export default Menu