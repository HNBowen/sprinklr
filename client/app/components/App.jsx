import React from 'react';

import Menu from './Menu.jsx'
import PlantList from './PlantList.jsx'

//dummy data for development, remove later
import { existingPlants, plantsToAdd } from '../../../dummyData.js'


class App extends React.Component {

  constructor(props) {
    super(props);

    this.state = {
      "addPlantModalVisible": false,
      "sort": "none",
      "plants": []
    }
  }

  handleOrderButtonClick() {

  }

  handleAddPlantButtonClick() {

  }

  handlePlantTileClick() {

  }

  render() {
    return (
        <div>
          <h1>Sprinklr</h1>
          <Menu handleOrderButtonClick={this.handleOrderButtonClick.bind(this)}
                handleAddPlantButtonClick={this.handleAddPlantButtonClick.bind(this)}
          />
          <PlantList plants={this.state.plants} handlePlantTileClick={this.handlePlantTileClick.bind(this)}/>
        </div>
      );
  }
}

export default App;