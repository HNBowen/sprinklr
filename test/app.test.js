import React from 'react'
import {shallow, mount} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import sinon from 'sinon';


import App from '../client/app/components/App.jsx'
import Menu from '../client/app/components/Menu.jsx'
import PlantList from '../client/app/components/PlantList.jsx'
import OrderButton from '../client/app/components/OrderButton.jsx'

import {existingPlants, plantsToAdd} from '../dummyData.js'


describe('App', function() {
  

  it('should render correctly', function() {
    const wrapper = shallow(<App />)

    jestExpect(shallowToJson(wrapper)).toMatchSnapshot();
  })

  //it should render a title h1 tag
  it('should render an h1 tag with text "Sprinklr"', function() {
    const wrapper = shallow(<App />)

    expect(wrapper.find('h1')).to.have.length(1);
    expect(wrapper.find('h1').text()).to.equal("Sprinklr");
  })

  //it should render 1 Menu element
  it('should render 1 Menu component', function() {
    const wrapper = shallow(<App />)
    expect(wrapper.find(Menu)).to.have.length(1);
  })

  //it should render 1 PlantList component
  it('should render 1 PlantList component', function() {
    const wrapper = shallow(<App />)
    expect(wrapper.find(PlantList)).to.have.length(1);
  })

  it('should render with state property "addPlantModalVisible" set to false', function() {
    const wrapper = shallow(<App />)
    expect(wrapper.state('addPlantModalVisible')).to.equal(false);
  })

  it('should render with state property "sort" set to false', function() {
    const wrapper = shallow(<App />)
    expect(wrapper.state('sort')).to.be.false
  })

  it('should render with state property "plants" and it should be an array', function() {
    const wrapper = shallow(<App />)
    const plants = wrapper.state("plants");
    expect(plants).to.not.equal(undefined)
    expect(Array.isArray(plants)).to.equal(true)
  })

  it('should have handleOrderButtonClick function', function() {
    const wrapper = shallow(<App />)
    const handler = wrapper.instance().handleOrderButtonClick;
    expect(typeof handler).to.equal("function");
  })

  it('should have handleAddPlantButtonClick function', function() {
    const wrapper = shallow(<App />)
    const handler = wrapper.instance().handleAddPlantButtonClick;
    expect(typeof handler).to.equal("function")
  })

  it('should have a handlePlantTileClick function', function() {
    const wrapper = shallow(<App />)
    const handler = wrapper.instance().handlePlantTileClick;
    expect(handler).to.be.a("function")
  })

  it('should pass plants array from state down to PlantList component', function() {
    const wrapper = shallow(<App />)
    wrapper.setState({"plants": existingPlants})
    const listWrapper = wrapper.find(PlantList);
    const listWrapperPlants = listWrapper.props()["plants"]
    expect(JSON.stringify(listWrapperPlants)).to.equal(JSON.stringify(existingPlants))

  })

  it('should pass sort state down to PlantList component', function() {
    const wrapper = shallow(<App />)
    const listWrapper = wrapper.find(PlantList);
    expect(listWrapper.props()["sort"]).to.be.false
  })

  it('should pass click handlers down to Menu component', function() {
    const wrapper = shallow(<App />);
    const menuWrapper = wrapper.find(Menu);
    expect(menuWrapper.props()["handleOrderButtonClick"]).to.be.a("function")
    expect(menuWrapper.props()["displayModal"]).to.be.a("function")
  })

  it('should pass handlePlantTileClick handler down to PlantList component', function() {
    const wrapper = shallow(<App />);
    const plantListWrapper = wrapper.find(PlantList);
    expect(plantListWrapper.props()["handlePlantTileClick"]).to.be.a("function")
  })

})

describe('handleOrderButtonClick', function() {

  it('should change App sort state to true on first click', function() {

    const wrapper = shallow(<App />);

    wrapper.instance().handleOrderButtonClick();

    const sort = wrapper.state()["sort"];
    expect(sort).to.be.true;
  })

  it('should change App sort state to false on second click', function() {
    const wrapper = shallow(<App />);

    wrapper.instance().handleOrderButtonClick();
    wrapper.instance().handleOrderButtonClick();

    const sort = wrapper.state()["sort"];
    expect(sort).to.be.false
  })
})

describe('handleAddPlantButtonClick', function() {

  it('should add a new plant object to the plants array', function() {

    const wrapper = shallow(<App />);

    wrapper.instance().handleAddPlantButtonClick(plantsToAdd[0]);

    expect(JSON.stringify(wrapper.state()["plants"][0])).to.equal(JSON.stringify(plantsToAdd[0]))
  })
})

describe('handlePlantTileClick', function() {

  it('should update the selected plant\'s lastWatered date', function() {

    const wrapper = shallow(<App />);
    wrapper.setState({"plants": existingPlants});
    const oldDate = existingPlants[0]["lastWatered"];
    wrapper.instance().handlePlantTileClick(1);
    const newDate = wrapper.state()["plants"][0]["lastWatered"];


    expect(oldDate < newDate).to.be.true;


  })
})

describe('displayModal', function() {

  it('should update addPlantModalVisible to true on first click', function() {

    const wrapper = shallow(<App />);
    wrapper.instance().displayModal();
    expect(wrapper.state()["addPlantModalVisible"]).to.be.true;
  })

  it('should change addPlantModalVisible back to false after second click', function() {

    const wrapper = shallow(<App />);

    wrapper.instance().displayModal();
    wrapper.instance().displayModal();
    expect(wrapper.state()["addPlantModalVisible"]).to.be.false
  })
})