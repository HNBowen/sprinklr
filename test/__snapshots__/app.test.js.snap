// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div>
  <h1>
    Sprinklr
  </h1>
  <Menu
    handleAddPlantButtonClick={[Function]}
    handleOrderButtonClick={[Function]}
  />
  <PlantList
    handlePlantTileClick={[Function]}
    plants={Array []}
  />
</div>
`;
